temporalio/converter.py

```diff
@@ -474,12 +474,21 @@ def encoding(self) -> str:
 
     def to_payload(self, value: Any) -> Optional[temporalio.api.common.v1.Payload]:
         """See base class."""
+        # Check for json callable, and invoke it
+        # This covers Pydantic models.
```

Comment:
My primary concern is we guarantee all of our serializable things work with lists, dictionaries, etc. What if I had a list of Pydantic models? The better way is to have full Pydantic JSON support through a custom converter which I need to write at https://github.com/temporalio/samples-python/issues/25. That would use `pydantic.json.pydantic_encoder` as mentioned at https://docs.pydantic.dev/usage/dataclasses/#json-dumping which would solve all issues. This is only solving some.

I wrote this class knowing it might be given a different encoder when instantiated. I will try to get to that sample soon. We may consider dropping/deprecating the limited Pydantic support we have in favor of a custom converter once I have that sample written.

Reply:
I thought its quite odd to send a List[PydanticModels] since Temporal recommend sending a single struct as params.. But its arguably correct too since its still a single struct.
 
I think having limited support is fine, if ppl want to use List[PydanticModels] then they can go for the custom converter? Otherwise the current support is enough to convert most of the use cases?

Writing it like this wont have any problem though, like in the test.
`class MyPydanticDTClassList(pydantic.BaseModel):
    foo_list: List[MyPydanticDTClass]`

Since am sending my payload from GO, if i use a custom converter, i need to create one for GO too to send the correct encoding type right? I think thats alot of effort, if we want to make cross language calling easier.

Anyway do you want me to drop this PR, since custom pydantic converter is probably the correct and best solution?

Reply:
> I think having limited support is fine, if ppl want to use List[PydanticModels] then they can go for the custom converter? Otherwise the current support is enough to convert most of the use cases?

Why not always go for custom converter if it's a short bit of code? Then you get guaranteed Pydantic support in all cases and you don't have to think about it. I will write it soon and if it seems easy enough to incorporate for users, will suggest that and deprecate/warn against built-in Pydantic use.

Reply:
Sure I will do the latter first, if performance is a concern. I will try the custom converter out to see if i can get it right.